src/components/UserSelector.tsx

```diff
@@ -1,34 +1,31 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
+import { useSelector } from 'react-redux';
+import { RootState } from '../app/store';
+import { setAuthor } from '../features/Author/authorSlice';
+import { useAppDispatch } from '../app/hooks';
 import { User } from '../types/User';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
+export const UserSelector: React.FC = ({}) => {
+  const dispatch = useAppDispatch();
+
+  const users = useSelector((state: RootState) => state.users);
```

Comment:
don't you use users only here? If yes delete usersSilce and keep data here, in state (or in App.tsx state - depends on your structure)